core/dbt/task/runnable.py

```diff
@@ -242,6 +242,11 @@ def execute_nodes(self):
                 raise
 
             for conn_name in adapter.cancel_open_connections():
+                if self.manifest is not None:
+                    node = self.manifest.nodes.get(conn_name)
+                    if node is not None and node.is_ephemeral_model:
```

Comment:
Are there any node types that won't have an `is_ephemeral_model` attribute? Like tests/snapshots/run-operations, etc?

Reply:
run-operations can't go through this path. Snapshots have `is_ephemeral_model`, this code path can only happen for nodes (in particular `manifest.nodes.get(conn_name) can only return an `Optional[CompileResultNode])` mypy will actually complain if `is_ephemeral_model` is missing from any type that can end up in `node`, I'm pretty sure!

Reply:
amazing, ship it :)